integration/ethereummock/node.go

```diff
@@ -63,8 +64,36 @@ type Node struct {
 	headOutCh chan *types.Block
 }
 
-func (m *Node) FetchBlock(id common.Hash) (*types.Block, bool) {
-	return m.Resolver.FetchBlock(id)
+// BlockListener is not used in the mock
+func (m *Node) BlockListener() chan *types.Header {
+	return make(chan *types.Header)
+}
+
+func (m *Node) FetchBlockByNumber(n *big.Int) (*types.Block, error) {
+	if n.Int64() == 0 {
+		return obscurocommon.GenesisBlock, nil
+	}
+	// TODO this should be a method in the resolver
+	var f bool
+	for blk, _ := m.Resolver.FetchHeadBlock(); blk.ParentHash() != obscurocommon.GenesisHash; {
+		if blk.Number() == n {
+			return blk, nil
+		}
+
+		blk, f = m.Resolver.FetchBlock(blk.ParentHash())
+		if !f {
+			return nil, fmt.Errorf("block in the chain without a parent")
+		}
+	}
+	return nil, nil
```

Comment:
@otherview I feel like the API's a bit confusing here, in that you can have a `nil` block returned, but also a `nil` error. It feels like anything that causes the block to be `nil` should have an error of some sort (e.g. `block not found` in this case).

Reply:
I brought this up a few days ago :) https://discord.com/channels/916052669955727371/945360340613484684/963083694736814120

I don't think `block not found` should be an error.  Since you correctly tried to fetch a value.
Because it's returning a pointer you will eventually have to check if that pointer is nil.
So returning `nil, nil` means that the fetch was successful and the element is nil. 

But this is opening a can of worms.
Tbh I just want to know which approach we're good with and stick to it :) 
@tudor-malene  what do you think ?